logback-core/src/main/java/ch/qos/logback/core/AsyncAppenderBase.java

```diff
@@ -152,12 +153,16 @@ private boolean isQueueBelowDiscardingThreshold() {
   }
 
   private void put(E eventObject) {
-    try {
-      blockingQueue.put(eventObject);
-    } catch (InterruptedException e) {
-      // Interruption of current thread when in doAppend method should not be consumed
-      // by AsyncAppender
-      Thread.currentThread().interrupt();  	
+    if (neverBlock) {
+      blockingQueue.offer(eventObject);
```

Comment:
Would it be safe to log a `WARN` status once (*not* a log message obviously, but a context status) when this happens?  Otherwise you would never know we lost your logs...